src/test/ui/consts/array-literal-index-oob.stderr

```diff
@@ -0,0 +1,18 @@
+error: index out of bounds: the len is 3 but the index is 4
+  --> $DIR/array-literal-index-oob.rs:2:7
+   |
+LL |     &{[1, 2, 3][4]};
+   |       ^^^^^^^^^^^^
+   |
+   = note: #[deny(const_err)] on by default
+
+error: reaching this expression at runtime will panic or abort
```

Comment:
I realize this is not caused by this PR, but isn't this error weird? "will unwind or abort" would make sense, and "will panic" also would, but "panic or abort" is a weird wording. Is there any case where it will *not* panic and somehow abort instead (as opposed to aborting *because* it panicked)?

Reply:
it will abort due to our promoted emergency abort code if we end up with a broken promoted that we can't turn back into runtime code because the user already expects a `&'static T`.

Reply:
That would be a bug in rustc though, right?

Reply:
probably, not sure if we managed to eliminate all promoted unevaluability. We could try moving the abort generation to just `bug!` out

Reply:
That can't work, this is a run-time check which we cannot do at compile-time.

Reply:
oh right...